From bfd5a709e74dfe9358b339b4a17c4ad6567b85c7 Mon Sep 17 00:00:00 2001
From: Tomohiro Kusumi <kusumi.tomohiro@gmail.com>
Date: Sun, 26 May 2019 05:42:09 +0900
Subject: [PATCH] Linux 5.2 compat: Directly call wait_on_page_bit()

wait_on_page_writeback() was made GPL only in torvalds/linux@19343b5bdd.

Directly call wait_on_page_bit() without using wait_on_page_writeback()
interface, given zfs_putpage() is the only caller for now.

Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Reviewed-by: loli10K <ezomori.nozomu@gmail.com>
Signed-off-by: Tomohiro Kusumi <kusumi.tomohiro@osnexus.com>
(backported from upstream commit bfd5a709e74dfe9358b339b4a17c4ad6567b85c7)
Signed-off-by: Colin Ian King <colin.king@canonical.com>
Closes #8794

Index: zfs-linux-0.7.12/module/zfs/zfs_vnops.c
===================================================================
--- zfs-linux-0.7.12.orig/module/zfs/zfs_vnops.c
+++ zfs-linux-0.7.12/module/zfs/zfs_vnops.c
@@ -4253,8 +4253,10 @@ zfs_putpage(struct inode *ip, struct pag
 		unlock_page(pp);
 		zfs_range_unlock(rl);
 
-		if (wbc->sync_mode != WB_SYNC_NONE)
-			wait_on_page_writeback(pp);
+		if (wbc->sync_mode != WB_SYNC_NONE) {
+			if (PageWriteback(pp))
+				wait_on_page_bit(pp, PG_writeback);
+		}
 
 		ZFS_EXIT(zfsvfs);
 		return (0);
